homeassistant/helpers/system_info.py

```diff
@@ -63,5 +84,9 @@ async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
             info_object["installation_type"] = "Home Assistant OS"
         else:
             info_object["installation_type"] = "Home Assistant Supervised"
+    elif "SUPERVISOR" in os.environ:
+        # hassio is not loaded, but we know we are running supervised
+        # because the SUPERVISOR env var is set
+        info_object["installation_type"] = "Home Assistant Supervised"
```

Comment:
Ideally we have a way to figure out if its HAOS in this case, but I couldn't come up with one

Reply:
I still feel like we should avoid this situation instead of trying to add some value that is maybe right.

Reply:
Will drop this block when I get back from lunch

Reply:
I was really hoping for a fallback way to detect HAOS but I haven't found a reliable one after a few hours of trying failed solution so ripping out is better

Reply:
f0675d61e2a073320d47c181199df0adfa1b7aba